File: backend/src/api/household.ts
import { Hono } from 'hono';
import { merge, preview } from '../skills/household/merge.js';

export const householdRoute = new Hono();

householdRoute.post('/preview', async (c) => {
  const body = await c.req.json<{ household_ids: string[] }>();
  return c.json(await preview(body));
});

householdRoute.post('/merge', async (c) => {
  const body = await c.req.json<{ household_ids: string[] }>();
  return c.json(await merge(body));
});

File: backend_py/stackwealth/api/household.py
"""/api/household — multi-household preview + merge."""
from __future__ import annotations

from fastapi import APIRouter, Request
from fastapi.responses import JSONResponse

from ..skills.household import merge, preview

router = APIRouter()


@router.post("/preview")
async def preview_route(request: Request) -> JSONResponse:
    b = await request.json()
    return JSONResponse(content=await preview(b))


@router.post("/merge")
async def merge_route(request: Request) -> JSONResponse:
    b = await request.json()
    return JSONResponse(content=await merge(b))
